### ISSUE TITLE
keypad in non-app mode assumes NumLock

### ISSUE THREAD

**Issue by kilobyte** [tomszilagyi/zutty#82]
While commit 15100bb5da8b26cb29c0137db2c0d946763535b9 fixed keypad in the app mode (thanks!), it regressed the arrow mode.  It now assumes that NumLock is always on, despite its true state, shift, and so on.

**Comment by tomszilagyi** [comment]
What do you mean by arrow mode, please give me a very specific example of what is wrong

**Comment by kilobyte** [comment]
> What do you mean by arrow mode

The default (ie, non-app) mode.  `\e>`

> please give me a very specific example of what is wrong

Start zutty, don't touch NumLock (ie, leave it off), press a kpad arrow.  You'll get a digit instead.

**Comment by tomszilagyi** [comment]
Non-app mode is numeric mode, and in that mode you get digits regardless of the NumLock state. That is what I get in xterm (vt220 keyboard) as well, which is what I try to emulate.

**Comment by tomszilagyi** [comment]
I can see a different regression: in application cursor key mode (DECAKM, CSI ?1h) the modifiers on normal (non keypad) arrows are discarded. I.e., after a `printf "\e[?1h" && cat -vT`, Alt+Left yields `SS3 D` instead of `CSI 1;3 D`. But that is orthogonal to keypad mode.

**Comment by kilobyte** [comment]
Xterm, like any other terminal emulator — and almost any other program — makes keypad arrows act same as "grey" arrows when NumLock is off, and same as numeric keys with NumLock on.

And most other terminals ignore NumLock state in app mode, heeding it only in the "normal" mode (that's how xterm's docs refers to DECKPNM).  This seems to be a quirk of X-Windows, I've just tried Linux console and PuTTY on Windows, they both behave that way.

**Comment by tomszilagyi** [comment]
Okay, maybe we are testing two different things. I am looking at Xterm-372 with VT220 keyboard mode enabled (this bit is important) on Linux as a reference.

If I test that with what you wrote here:
> don't touch NumLock (ie, leave it off), press a kpad arrow. You'll get a digit instead.

... I get a digit.

If I put the terminal in application keypad mode, e.g., `printf "\e=" && cat -vT` then I get different encodings depending on NumLock, Shift, etc. I tried to summarize this at [the updated docs](https://tomscii.sig7.se/zutty/doc/KEYS.html#Sequences%20dependent%20on%20keypad%20mode%20setting)

I thought terminal apps set application keypad mode when they intend to interpret the keypad buttons as arrows, and turn that off (set numeric mode) when they want to see numbers.

**Comment by kilobyte** [comment]
Do you have some weird xresources?  I tried a bunch of terminals, some that recognize application mode and some that don't, and not a single one forces numlock to either state in the "normal" mode.

**Comment by tomszilagyi** [comment]
```
Xcursor.size:                  24
XTerm*background:              black
XTerm*foreground:              white
XTerm*vt100.utf8:              true
XTerm*vt100.utf8Title:         true
XTerm*vt100.initialFont:       5
XTerm*vt100.saveLines:         10000
XTerm*keyboardType:            vt220
XTerm*keepSelection:           false
XTerm*selectToClipboard:       false
XTerm*metaSendsEscape:         true
XTerm*modifyOtherKeys:         1
XTerm*alternateScroll:         true

XTerm*vt100*translations:      #override \n\
                               Ctrl Shift <Key>C:  copy-selection(CLIPBOARD) \n\
                               Ctrl Shift <Key>V:  insert-selection(CLIPBOARD)
```

**Comment by tomszilagyi** [comment]
> I can see a different regression: in application cursor key mode (DECAKM, CSI ?1h) the modifiers on normal (non keypad) arrows are discarded. I.e., after a `printf "\e[?1h" && cat -vT`, Alt+Left yields `SS3 D` instead of `CSI 1;3 D`. But that is orthogonal to keypad mode.

Fixed in e1687aa

**Comment by tomszilagyi** [comment]
I don't see anything to be fixed here, closing.
